### ISSUE TITLE
Decorations: codicon icons appear quite large

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#133422]
Editor labels now use the lock icon to indicate readonly state and the icon is quite large:

![image](https://user-images.githubusercontent.com/900690/133960600-1b2d3023-3b2c-4c36-a0e9-7a2ca81e2927.png)

This seems to be using a font size of 16px, while normal decorations are at 90% of our default font size (13px).

I understand we cannot use the same size to make icons appear nicely, however 16px seems quite large.

Here is 14px as comparison:

![image](https://user-images.githubusercontent.com/900690/133960698-cea742ee-6594-421d-a3b3-1456730ec396.png)

I think this changed with https://github.com/microsoft/vscode/commit/aa454714b70a18cc5b3e1aa392e83bda114af735 so assigning to Daniel.

**Comment by Tyriar** [comment]
The icons were designed to be at least 16px so they are really blurry when smaller on 1.0 device pixel ratio displays.

**Comment by misolori** [comment]
I'd also be in favor of adjusting the font-size to an actual pixel instead of a percent since that has caused issues with alignment within decorators.

**Comment by Tyriar** [comment]
@misolori i think anything less than 16 will cause all lines to be blurry, these lower dpi displays are still capture most users I believe.

**Comment by gjsjohnmurray** [comment]
Experimenting for #134121, here's how the readonly indicator looks at 12px. I have put the tab into a locked group for comparison.

![image](https://user-images.githubusercontent.com/6726799/135297376-a828b316-a098-416e-ba5e-fc19d4a1ecb4.png)

Perhaps this makes it less likely that @isidorn will want to click it 😄

**Comment by bpasero** [comment]
I think this also causes the pin icon to appear larger than it used to be:

![image](https://user-images.githubusercontent.com/900690/135298197-21ca4a90-2294-4a1c-bda9-33b119edf870.png)

**Comment by bpasero** [comment]
Yeah I was about to suggest the same, only have small variants for the ones we use, maybe same name but suffixed `-small`?
